### ISSUE TITLE
Date filtering seems to be unimplemented

### ISSUE THREAD

**Issue by nathan-mill** [swellstores/swellpy#6]
I'm following the examples in the integration tests and calling the Swell object as follows (`self.api` is an instance of `Swell`):

```
EXPAND_NESTED_PROPERTIES = [
    "items.bundle_items.product",
    "items.product",
    "shipments:1000",
]

response = self.api.orders.list(
    {"date_created": {"$gte": timestamp}, "expand": EXPAND_NESTED_PROPERTIES}
)
```

The URL which swellpy generates from this looks like this:

```
https://mill-dev:<redacted>@api.swell.store/orders?date_created=%24gte&expand=items.bundle_items.product&expand=items.product&expand=shipments%3A1000
```

As you can see, the timestamp is never actually inserted anywhere in the query string and the result of any call with this malformed `date_created` value is always empty.

It seems like every integration test related to date filtering is simply looking for an empty result, which does not exactly prove that date filtering works, it just proves that an empty result set is returned.

**Comment by nathan-mill** [comment]
Stepping through the library in `pdb` it seems like that nested `params` dictionary is just being passed down to `requests` and I don't believe that `requests` knows how to handle nested objects like that on its own.

**Comment by logeshswell** [comment]
@nathan-mill Yes, you are totally right. `requests` don't know how to handle nested objects.

I created a [pull request](https://github.com/swellstores/swellpy/pull/7) for the fix. Please refer to the code changes for the time-being.

**Comment by nathan-mill** [comment]
Thanks, but based on my testing, I don't believe that will be sufficient.

The Swell backend appears to be expecting all requests to conform to the non-standard approach of the `qs` NPM package when it comes to query strings. There is absolutely nothing in Python (or any other language I'm aware of) that is going to generate the same output as `qs`. From what I have seen first-hand, any failure to conform to the `qs` format at all results in empty response pages. 

Your own Node package eschews this requirement by seemingly opening up a raw socket and piping Mongo queries directly into it, but if you simply want to make traditional HTTP requests, it seems like it's necessary to re-implement all of `qs` in other languages.

**Comment by ericingram** [comment]
If I recall the qs library worked the same way PHP query parsing used to work, because our first SDK was for PHP years ago. Surprised nothing does this for Python though.

**Comment by nathan-mill** [comment]
Everything in modern Python strives to comply with RFCs related to proper URI formation. qs has never been even remotely compliant and there's really no demand for it outside the NPM ecosystem so adopting it excludes quite a bit of the world. 

I believe there is _some_ degree of overlap between https://www.php.net/manual/en/function.http-build-query.php and what qs does, but qs gets a lot more convoluted. They're not completely equivalent. 

I think this library either needs to adopt the same usage pattern as the Node version in opening up a socket to stream raw Mongo queries through or it just shouldn't be public. In its current state, it's really not compatible with very much of the Swell backend and cannot be used.

**Comment by ericingram** [comment]
That’s fair. We can probably support a header to parse the query string using a more standard approach @logeshswell
